File: server/src/models/Workspace.model.ts
import mongoose, { Schema } from 'mongoose';

const workspaceSchema = new Schema(
  {
    name: { type: String, required: true },
    slug: { type: String, required: true, unique: true },
    description: { type: String, default: '' },
    createdBy: { type: Schema.Types.ObjectId, ref: 'User' },
  },
  { timestamps: true }
);

export const WorkspaceModel = mongoose.model('Workspace', workspaceSchema);

File: server-go/internal/modules/workspace/workspace_model.go
package workspace

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Workspace struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
	Name        string             `bson:"name" json:"name"`
	Slug        string             `bson:"slug" json:"slug"`
	Owner       primitive.ObjectID `bson:"owner" json:"owner"`
	CreatedBy   primitive.ObjectID `bson:"createdBy,omitempty" json:"createdBy,omitempty"`
	Description string             `bson:"description" json:"description"`
	CreatedAt   time.Time          `bson:"createdAt" json:"createdAt"`
	UpdatedAt   time.Time          `bson:"updatedAt" json:"updatedAt"`
}

type WorkspaceMember struct {
	ID        primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
	Workspace primitive.ObjectID `bson:"workspace" json:"workspace"`
	User      primitive.ObjectID `bson:"user" json:"user"`
	Role      string             `bson:"role" json:"role"`
	CreatedAt time.Time          `bson:"createdAt" json:"createdAt"`
	UpdatedAt time.Time          `bson:"updatedAt" json:"updatedAt"`
}
